shell/common/shell.cc

```diff
@@ -2023,6 +2037,71 @@ bool Shell::OnServiceProtocolReloadAssetFonts(
   return true;
 }
 
+void Shell::AddView(int64_t view_id) {
+  TRACE_EVENT0("flutter", "Shell::AddView");
+  FML_DCHECK(is_set_up_);
+  FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
+  if (view_id == kFlutterImplicitViewId) {
+    FML_DLOG(WARNING)
+        << "Unexpected request to add the implicit view #"
+        << kFlutterImplicitViewId
+        << ". This view should never be added. This call is no-op.";
+    return;
+  }
+
+  task_runners_.GetUITaskRunner()->PostTask([engine = engine_->GetWeakPtr(),  //
+                                             view_id                          //
+  ] {
+    if (engine) {
+      engine->AddView(view_id);
```

Comment:
Why is the view added to the UI thread before the raster thread? Wouldn't that mean the UI thread might attempt to render to a view before the raster thread has added that view? Should we swap this order such that the view is added to the rasterizer before the engine?

Reply:
I've revised the blocking logic in `Shell::{Add,Remove}View` with some docs as well. Let me know what you think!